design-proposals/garbage-collect-completed-dvs.md

```diff
@@ -0,0 +1,61 @@
+# Overview
+DataVolumes are a useful construct for provisioning VM disks but once the PVC is provisioned and populated the DataVolume provides no value. This design presents an approach for garbage collecting completed DataVolumes. 
+
+## Motivation
+In the field, customers are confused by DVs after the fact. For example, when someone deletes a PVC that is owned by a DV they are confused when the PVC is re-created.
```

Comment:
Additional motivations:
* In kubevirt we have established a convention that DataVolumes are for **provisioning** VM disks and PVCs are for **consuming** VM disks.  Still, sometimes people are tempted to misuse a DV for a consumption-related task (ie. to resize the disk or to check the current size of the disk).  Confusion also arises when objects are deleted.  Deleting a DV should effectively cancel a provisioning operation (also deleting the unprepared PVC).  When a user deletes a prepared PVC the intent is usually to delete the disk being used but with today's behavior, the PVC gets recreated by the DV controller.
So the overall motivation of this feature is to make the API objects defined in the cluster codify the convention of a DV representing a disk provisioning operation and a PVC representing the disk itself.

Reply:
Adding.